
]
[2023-07-20 12:23:39,174: ERROR: main: >>>>> Stage | Data Ingestion Stage | Failed]
Traceback (most recent call last):
  File "main.py", line 12, in <module>
    obj.main()
  File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\Admin\anaconda3\envs\mlops\lib\zipfile.py", line 1269, in __init__
    self._RealGetContents()
  File "C:\Users\Admin\anaconda3\envs\mlops\lib\zipfile.py", line 1336, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2023-07-20 12:30:38,526: INFO: main: >>>>> Stage | Data Ingestion Stage | Started]
[2023-07-20 12:30:38,531: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 12:30:38,533: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 12:30:38,533: INFO: common: created directory at: artifacts]
[2023-07-20 12:30:38,534: INFO: common: created directory at: artifacts/data_ingestion]
[2023-07-20 12:30:39,041: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Content-Type: text/html; charset=utf-8
Date: Thu, 20 Jul 2023 10:30:38 GMT
Cache-Control: no-cache, no-store
Pragma: no-cache
Set-Cookie: ka_sessionid=4521c2caaaad181557b710349f46bf1f; max-age=2626560; path=/
Set-Cookie: CSRF-TOKEN=CfDJ8MCGpmbK0d9IuUUyozUQ8_km1MQOxu3gdTwmEllA416OCPfQHC4ng00Z6PuLOi2Ih1hsZoxTWS8BULPCjEIJHIzOuWGZwxI0y03iDk2URw; path=/; secure; samesite=lax; httponly
Set-Cookie: XSRF-TOKEN=CfDJ8MCGpmbK0d9IuUUyozUQ8_nIKh7tiNOOSjamWZbB-LkUh323AJBYL2zmyl72tTXFJJTQmJFbDz45zFXMy-FvhbSq_cldTK4V-p-tpVYsYzXekQ; path=/; secure; samesite=lax
Set-Cookie: CLIENT-TOKEN=eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJpc3MiOiJrYWdnbGUiLCJhdWQiOiJjbGllbnQiLCJzdWIiOm51bGwsIm5idCI6IjIwMjMtMDctMjBUMTA6MzA6MzkuMDM3MzE5MloiLCJpYXQiOiIyMDIzLTA3LTIwVDEwOjMwOjM5LjAzNzMxOTJaIiwianRpIjoiMWZhZWZlZjAtZjllNy00NWUyLWFkYmUtZmE1YTgzODEyNzViIiwiZXhwIjoiMjAyMy0wOC0yMFQxMDozMDozOS4wMzczMTkyWiIsImFub24iOnRydWUsImZmIjpbIktlcm5lbHNGaXJlYmFzZUxvbmdQb2xsaW5nIiwiQ29tbXVuaXR5TG93ZXJIZWFkZXJTaXplcyIsIkFsbG93Rm9ydW1BdHRhY2htZW50cyIsIkZvcmNlTmF2Rm9vdGVyT25Nb2JpbGVBbmRUYWJsZXQiLCJGcm9udGVuZEVycm9yUmVwb3J0aW5nIiwiQ29tcGV0aXRpb25zUHl0aG9uTWV0cmljcyIsIkRhdGFzZXRzTWFuYWdlZEZvY3VzT25PcGVuIiwiRG9pRGF0YXNldFRvbWJzdG9uZXMiLCJDaGFuZ2VEYXRhc2V0T3duZXJzaGlwVG9PcmciLCJLZXJuZWxFZGl0b3JIYW5kbGVNb3VudE9uY2UiLCJLTUxvZ2dlZE91dEhvbWVwYWdlIiwiUGFnZVZpc2liaWxpdHlBbmFseXRpY3NSZXBvcnRlciIsIk1hdVJlcG9ydCIsIk1vZGVsc0NhY2hlZFRhZ1NlcnZpY2VFbmFibGVkIiwiQ29tcGV0aXRpb25zUnVsZXNLbSIsIkRhdGFzZXRzU2hhcmVkV2l0aFRoZW1TZWFyY2giLCJEYXRhc2V0c1ZvdGluZ0NoaXBzIiwiRGF0YXNldHNGZWVkYmFja3NCdXR0b25OdWRnZSIsIkRhdGFzZXRzRmVlZGJhY2tzRGVsYXlOdWRnZSIsIlRhZ3NNZW51U2xpZGVyUGFuZWwiLCJNb2RlbEluc3RhbmNlUmVuZGVyZWRVc2FnZSIsIlJlY2VudGx5Vmlld2VkTW9kZWxzU2hlbGYiLCJLbUVtb2ppUGlja2VyIiwiS01SYW5raW5ncyJdLCJmZmQiOnsiS2VybmVsRWRpdG9yQXV0b3NhdmVUaHJvdHRsZU1zIjoiMzAwMDAiLCJGcm9udGVuZEVycm9yUmVwb3J0aW5nU2FtcGxlUmF0ZSI6IjAiLCJFbWVyZ2VuY3lBbGVydEJhbm5lciI6Int9IiwiQ2xpZW50UnBjUmF0ZUxpbWl0IjoiNDAiLCJGZWF0dXJlZENvbW11bml0eUNvbXBldGl0aW9ucyI6IjM1MzI1LDM3MTc0LDMzNTc5LDM3ODk4LDM3MzU0LDM3OTU5LDM4ODYwIiwiQWRkRmVhdHVyZUZsYWdzVG9QYWdlTG9hZFRhZyI6ImRpc2FibGVkIiwiQ29tcGV0aXRpb25NZXRyaWNUaW1lb3V0TWludXRlcyI6IjMwIn0sInBpZCI6ImthZ2dsZS0xNjE2MDciLCJzdmMiOiJ3ZWItZmUiLCJzZGFrIjoiQUl6YVN5QTRlTnFVZFJSc2tKc0NaV1Z6LXFMNjU1WGE1SkVNcmVFIiwiYmxkIjoiOWQ1YjE4YjNhYTczOGFiYzg4N2NkMTFjNmEyNjM1NjVhYzM3MWY1ZSJ9.; path=/; secure; samesite=lax
Transfer-Encoding: chunked
Vary: Accept-Encoding
Turbolinks-Location: https://www.kaggle.com/account/login?titleType=dataset-downloads&showDatasetDownloadSkip=False&messageId=datasetsWelcome&returnUrl=%2Fdatasets%2Fnkifor%2Fpneumonia-dataset%2Fversions%2F1%3Fresource%3Ddownload
X-Frame-Options: SAMEORIGIN
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Content-Security-Policy: object-src 'none'; script-src 'nonce-pp2iVzsXgNxF6HZ80/2XZA==' 'report-sample' 'unsafe-inline' 'unsafe-eval' 'strict-dynamic' https: http:; base-uri 'none'; report-uri https://csp.withgoogle.com/csp/kaggle/20201130; frame-src 'self' https://www.kaggleusercontent.com https://www.youtube.com/embed/ https://polygraph-cool.github.io https://www.google.com/recaptcha/ https://form.jotform.com https://submit.jotform.us https://submit.jotformpro.com https://submit.jotform.com https://www.docdroid.com https://www.docdroid.net https://kaggle-static.storage.googleapis.com https://kkb-production.jupyter-proxy.kaggle.net https://kkb-production.firebaseapp.com https://kaggle-metastore.firebaseapp.com https://apis.google.com https://content-sheets.googleapis.com/ https://accounts.google.com/ https://storage.googleapis.com https://docs.google.com https://drive.google.com https://calendar.google.com/;
X-Content-Type-Options: nosniff
Referrer-Policy: strict-origin-when-cross-origin
Via: 1.1 google
Set-Cookie: GCLB=CLaLhO_-zbOBfg; path=/; HttpOnly
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Connection: close

]
[2023-07-20 12:30:39,053: ERROR: main: >>>>> Stage | Data Ingestion Stage | Failed]
Traceback (most recent call last):
  File "main.py", line 12, in <module>
    obj.main()
  File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\Admin\anaconda3\envs\mlops\lib\zipfile.py", line 1269, in __init__
    self._RealGetContents()
  File "C:\Users\Admin\anaconda3\envs\mlops\lib\zipfile.py", line 1336, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2023-07-20 16:33:33,931: INFO: main: >>>>> Stage | Data Ingestion Stage | Started]
[2023-07-20 16:33:33,943: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 16:33:33,946: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 16:33:33,947: INFO: common: created directory at: artifacts]
[2023-07-20 16:33:33,947: INFO: common: created directory at: artifacts/data_ingestion]
[2023-07-20 16:33:58,428: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
x-amz-id-2: wJfJffVFTlhG2w/SHto9Pr8ruXiY3vZsTU9PIx27kRG++vvqkyvYHzBv87AX6O9EMYB6blj0Y0A=
x-amz-request-id: KG3DSW4Q589TW81P
Date: Thu, 20 Jul 2023 14:33:35 GMT
Last-Modified: Thu, 20 Jul 2023 12:17:51 GMT
ETag: "1dd1f6e6ea0cae949a176df68fb56036-67"
x-amz-server-side-encryption: AES256
Accept-Ranges: bytes
Content-Type: application/zip
Server: AmazonS3
Content-Length: 1145020290
Connection: close

]
[2023-07-20 16:34:08,731: INFO: main: >>>>> Stage | Data Ingestion Stage | Finished 

 ========================================]
[2023-07-20 16:34:08,732: INFO: main: *******************]
[2023-07-20 16:34:08,732: INFO: main: >>>>>> Stage | Prepare base model | Started <<<<<<]
[2023-07-20 16:34:08,736: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 16:34:08,738: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 16:34:08,738: INFO: common: created directory at: artifacts]
[2023-07-20 16:34:08,739: INFO: common: created directory at: artifacts/prepare_base_model]
[2023-07-20 16:34:09,240: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-07-20 16:34:09,431: INFO: main: >>>>>> Stage Prepare base model | Finished <<<<<<

x==========x]
[2023-07-20 16:34:09,432: INFO: main: *******************]
[2023-07-20 16:34:09,432: INFO: main: >>>>>> Stage | Training Stage | Started <<<<<<]
[2023-07-20 16:34:09,435: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 16:34:09,437: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 16:34:09,438: INFO: common: created directory at: artifacts]
[2023-07-20 16:34:09,439: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2023-07-20 16:34:09,440: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2023-07-20 16:34:09,442: INFO: common: created directory at: artifacts\training]
[2023-07-20 16:34:12,668: ERROR: main: >>>>> Stage | Training Stage | Failed]
Traceback (most recent call last):
  File "main.py", line 37, in <module>
    model_trainer.main()
  File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\pipeline\stage_03_training.py", line 25, in main
    training.train(
  File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\components\training.py", line 71, in train
    self.model.fit(
  File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\utils\traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\tensorflow\python\eager\execute.py", line 53, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "main.py", line 37, in <module>
      model_trainer.main()
    File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\pipeline\stage_03_training.py", line 25, in main
      training.train(
    File "c:\edu\python\mlops_projects\mlops_pneumonia_xray_detector\src\CNN_pneumonia_xray_detector\components\training.py", line 71, in train
      self.model.fit(
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\training.py", line 1742, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\training.py", line 1338, in train_function
      return step_function(self, iterator)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\training.py", line 1322, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\training.py", line 1303, in run_step
      outputs = model.train_step(data)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\training.py", line 1081, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\training.py", line 1139, in compute_loss
      return self.compiled_loss(
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\losses.py", line 142, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\losses.py", line 268, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\losses.py", line 2122, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\Admin\anaconda3\envs\mlops\lib\site-packages\keras\src\backend.py", line 5566, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,2] labels_size=[16,3]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_1478]
[2023-07-20 17:16:39,693: INFO: main: >>>>> Stage | Data Ingestion Stage | Started]
[2023-07-20 17:16:39,696: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 17:16:39,698: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 17:16:39,699: INFO: common: created directory at: artifacts]
[2023-07-20 17:16:39,699: INFO: common: created directory at: artifacts/data_ingestion]
[2023-07-20 17:17:06,185: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
x-amz-id-2: xQVEM/bwt6J/49paeVo5U7M++vioWLAF1VlDQp22OTQDO2ihlzjl/HkyhrQfENPIv2aC/4vzolA=
x-amz-request-id: 95WS39FCNCBZDFZM
Date: Thu, 20 Jul 2023 15:16:41 GMT
Last-Modified: Thu, 20 Jul 2023 12:17:51 GMT
ETag: "1dd1f6e6ea0cae949a176df68fb56036-67"
x-amz-server-side-encryption: AES256
Accept-Ranges: bytes
Content-Type: application/zip
Server: AmazonS3
Content-Length: 1145020290
Connection: close

]
[2023-07-20 17:17:16,410: INFO: main: >>>>> Stage | Data Ingestion Stage | Finished 

 ========================================]
[2023-07-20 17:17:16,410: INFO: main: *******************]
[2023-07-20 17:17:16,410: INFO: main: >>>>>> Stage | Prepare base model | Started <<<<<<]
[2023-07-20 17:17:16,427: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 17:17:16,429: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 17:17:16,429: INFO: common: created directory at: artifacts]
[2023-07-20 17:17:16,430: INFO: common: created directory at: artifacts/prepare_base_model]
[2023-07-20 17:17:16,784: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-07-20 17:17:16,973: INFO: main: >>>>>> Stage Prepare base model | Finished <<<<<<

x==========x]
[2023-07-20 17:17:16,973: INFO: main: *******************]
[2023-07-20 17:17:16,974: INFO: main: >>>>>> Stage | Training Stage | Started <<<<<<]
[2023-07-20 17:17:16,977: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 17:17:16,979: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 17:17:16,979: INFO: common: created directory at: artifacts]
[2023-07-20 17:17:16,980: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2023-07-20 17:17:16,981: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2023-07-20 17:17:16,983: INFO: common: created directory at: artifacts\training]
[2023-07-20 17:37:05,731: INFO: main: >>>>>> Stage | Training Stage | Finished <<<<<<

x==========x]
[2023-07-20 17:37:05,731: INFO: main: *******************]
[2023-07-20 17:37:05,731: INFO: main: >>>>>> Stage | Evaluation stage | Started <<<<<<]
[2023-07-20 17:37:05,741: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-07-20 17:37:05,743: INFO: common: yaml file: params.yaml loaded successfully]
[2023-07-20 17:37:05,743: INFO: common: created directory at: artifacts]
[2023-07-20 17:38:12,045: INFO: common: json file saved at: scores.json]
[2023-07-20 17:38:12,045: INFO: main: >>>>>> Stage Evaluation stage Completed <<<<<<

x==========x]
